x-pack/plugins/reporting/public/directives/export_config/export_config.js

```diff
@@ -104,6 +104,13 @@ module.directive('exportConfig', ($rootScope, reportingDocumentControl, reportin
               });
             }
 
+            if (err.status === 403) {
+              return toastNotifications.addDanger({
+                title: 'Reporting error',
+                text: `You don't have permission to generate this report.`,
+              });
```

Comment:
I think you'll have to add in `'data-test-subj': 'permissionReportError'` and update the failing test so it now makes sure you get this error.  Otherwise, LGTM!